torch/nn/parallel/distributed.py

```diff
@@ -1330,9 +1204,7 @@ def _passing_sync_batchnorm_handle(self, module_copies):
                     assert (
                         self.device_type != "cpu"
                     ), "SyncBatchNorm layers only work with GPU modules"
-                    layer._specify_ddp_gpu_num(
-                        len(self.device_ids) if self.device_ids else 1
-                    )
+                    layer._specify_ddp_gpu_num(1)
```

Comment:
I wonder if we can always hardcode this to 1 now in sync batchnorm. Looking at the SyncBN implementation, IIUC it looks like setting this attribute is what lets the layer know we are training in DDP (which isn't very robust either since anyone can set it it seems like). So we would need to figure out some way to ensure that SyncBN is only used with DDP if we decide that we would like to remove these calls.

Reply:
Thanks for the suggestion! Plan to remove `_specify_ddp_gpu_num` method later, as it is also used by fairscale.

Reply:
Resolved by #56425 and #55946.